Dev with azion pt br (#332)
aziontech/docs: src/content/docs/pt-br/pages/develop-with-azion-pt/lang-wasm/lang-wasm.mdx

        0, 97, 115, 109, 1, 0, 0, 0, 1, 6, 1, 96, 1, 127, 1, 127, 3, 2, 1, 0, 7, 13, 1, 9, 105, 110, 99, 114, 101,
        109, 101, 110, 116, 0, 0, 10, 9, 1, 7, 0, 32, 0, 65, 1, 106, 11,
      ]);

      const wasmModule = new WebAssembly.Module(wasmCode);
      const wasmInstance = new WebAssembly.Instance(wasmModule);
      const increment = wasmInstance.exports.increment;
      let value = 0;

      if (request.headers.has("value")) {
        let number = request.headers.get("value");
        if (!isNaN(number)) {
          value = parseInt(number);
        }
      }
      let incremented = increment(value);
      return new Response(
        "incrementing " + value + " we have " + incremented
      );
    }
    addEventListener("fetch", (event) => {
      event.respondWith(handleRequest(event.request));
    });
```

A constante `wasmCode` é a representação binária do seguinte [código WAT](https://developer.mozilla.org/en-US/docs/WebAssembly/Understanding_the_text_format):

```
    (module
    (type (;0;) (func (param i32) (result i32)))
    (func (;0;) (type 0) (param i32) (result i32)
      local.get 0
      i32.const 1
      i32.add)
    (export “increment” (func 0)))
```

> > **Nota**: um [código WAT](https://developer.mozilla.org/en-US/docs/WebAssembly/Understanding_the_text_format) é a representação textual do formato binário do wasm, usado para permitir que o WebAssembly seja lido e editado por humanos.

Reviewer: ```suggestion
:::note
Um [código WAT](https://developer.mozilla.org/en-US/docs/WebAssembly/Understanding_the_text_format) é a representação textual do formato binário do wasm, usado para permitir que o WebAssembly seja lido e editado por humanos.
:::
```

6. Clique no botão **Save**

> Agora, a função está pronta para ser instanciada em uma edge application. Veja como [instanciar e executar as funções em sua edge application](/pt-br/documentacao/produtos/edge-application/edge-functions-instances/).

Reviewer: ```suggestion
Agora, a função está pronta para ser instanciada em uma edge application. Veja como [instanciar e executar as funções em sua edge application](/pt-br/documentacao/produtos/guias/build/instanciar-edge-functions/).
```

---

> title: Como construir funçöes no edge com WebAssembly

Reviewer 2: ```suggestion
title: Como construir funções no edge com WebAssembly
```

description: Aprenda sobre o processo necessário para a criação de uma edge function que utiliza uma função gerada por WebAssembly.
meta_tags: javascript, edge computing, wasm, webassembly
namespace: documentation_dev_with_azion_lang_wasm
permalink: /documentacao/produtos/build/desenvolvimento-azion/linguagens/wasm/
menu_namespace: buildMenu
---

Saiba mais sobre o processo necessário para a criação de uma [edge function](/pt-br/documentacao/produtos/edge-application/edge-functions/) que utiliza uma função originada de um arquivo [WebAssembly](https://webassembly.org/getting-started/developers-guide/).

> 1. No [Real-Time Manager (RTM)](https://manager.azion.com/), no canto superior esquerdo, selecione **Edge Functions** na seção **Edge Libraries**.
> 2. Clique em **Add a Function**.
> 3. Escolha um nome para sua função.
> 4. Em **Language**, selecione **JavaScript**.
> 5. Copie e cole o seguinte código de exemplo no bloco de código:

Reviewer 2: ```suggestion
1. No [Real-Time Manager (RTM)](https://manager.azion.com/), no canto superior esquerdo, selecione o **Products menu**, representado por três linhas horizontais.
2. Selecione **Edge Functions** na seção **Edge Libraries**.
3. Clique em **Add a Function**.
4. Escolha um nome para sua função.
5. Em **Language**, selecione **JavaScript**.
6. Copie e cole o seguinte código de exemplo no bloco de código:
```

        109, 101, 110, 116, 0, 0, 10, 9, 1, 7, 0, 32, 0, 65, 1, 106, 11,
      ]);

      const wasmModule = new WebAssembly.Module(wasmCode);
      const wasmInstance = new WebAssembly.Instance(wasmModule);
      const increment = wasmInstance.exports.increment;
      let value = 0;

      if (request.headers.has("value")) {
        let number = request.headers.get("value");
        if (!isNaN(number)) {
          value = parseInt(number);
        }
      }
      let incremented = increment(value);
      return new Response(
        "incrementing " + value + " we have " + incremented
      );
    }
    addEventListener("fetch", (event) => {
      event.respondWith(handleRequest(event.request));
    });
```

A constante `wasmCode` é a representação binária do seguinte [código WAT](https://developer.mozilla.org/en-US/docs/WebAssembly/Understanding_the_text_format):

```
    (module
    (type (;0;) (func (param i32) (result i32)))
    (func (;0;) (type 0) (param i32) (result i32)
      local.get 0
      i32.const 1
      i32.add)
    (export “increment” (func 0)))
```

:::note

> Um [código WAT](https://developer.mozilla.org/en-US/docs/WebAssembly/Understanding_the_text_format) é a representação textual do formato binário do wasm, usado para permitir que o WebAssembly seja lido e editado por humanos.

Reviewer 3: ```suggestion
Um [código WAT](https://developer.mozilla.org/en-US/docs/WebAssembly/Understanding_the_text_format) é a representação textual do formato binário do Wasm, usado para permitir que o WebAssembly seja lido e editado por humanos.
```

:::

> 
> 

Reviewer 2: ```suggestion
```

> 6. Clique no botão **Save**

Reviewer 3: ```suggestion
6. Clique no botão **Save**.
```
